src/metadata/trackmetadatataglib.cpp

```diff
@@ -0,0 +1,688 @@
+    }
+
+    if (tag.fieldListMap().contains("REPLAYGAIN_ALBUM_GAIN")) {
+        pTrackMetadata->setReplayGainDbString(
+                toQStringFirst(tag.fieldListMap()["REPLAYGAIN_ALBUM_GAIN"]));
+    }
+    //Prefer track gain over album gain.
+    if (tag.fieldListMap().contains("REPLAYGAIN_TRACK_GAIN")) {
+        pTrackMetadata->setReplayGainDbString(
+                toQStringFirst(tag.fieldListMap()["REPLAYGAIN_TRACK_GAIN"]));
+    }
+
+    /*
+     * Reading key code information
+     * Unlike, ID3 tags, there's no standard or recommendation on how to store 'key' code
+     *
+     * Luckily, there are only a few tools for that, e.g., Rapid Evolution (RE).
+     * Assuming no distinction between start and end key, RE uses a "INITIALKEY"
+     * or a "KEY" vorbis comment.
+     */
+    if (tag.fieldListMap().contains("KEY")) {
+        pTrackMetadata->setKey(toQStringFirst(tag.fieldListMap()["KEY"]));
+    }
+    if (pTrackMetadata->getKey().isEmpty()
+            && tag.fieldListMap().contains("INITIALKEY")) {
+        // try alternative field name
+        pTrackMetadata->setKey(
+                toQStringFirst(tag.fieldListMap()["INITIALKEY"]));
+    }
+
+    if (tag.fieldListMap().contains("ALBUMARTIST")) {
+        pTrackMetadata->setAlbumArtist(
+                toQStringFirst(tag.fieldListMap()["ALBUMARTIST"]));
+    }
+    if (pTrackMetadata->getAlbumArtist().isEmpty()
+            && tag.fieldListMap().contains("ALBUM_ARTIST")) {
+        // try alternative field name
+        pTrackMetadata->setAlbumArtist(
+                toQStringFirst(tag.fieldListMap()["ALBUM_ARTIST"]));
+    }
+    if (pTrackMetadata->getAlbumArtist().isEmpty()
+            && tag.fieldListMap().contains("ALBUM ARTIST")) {
+        // try alternative field name
+        pTrackMetadata->setAlbumArtist(
+                toQStringFirst(tag.fieldListMap()["ALBUM ARTIST"]));
+    }
+
+    if (tag.fieldListMap().contains("COMPOSER")) {
+        pTrackMetadata->setComposer(
+                toQStringFirst(tag.fieldListMap()["COMPOSER"]));
+    }
+
+    if (tag.fieldListMap().contains("GROUPING")) {
+        pTrackMetadata->setGrouping(
+                toQStringFirst(tag.fieldListMap()["GROUPING"]));
+    }
+
+    if (tag.fieldListMap().contains("DATE")) {
+        pTrackMetadata->setYear(toQStringFirst(tag.fieldListMap()["DATE"]));
+    }
+}
+
+void readMP4Tag(TrackMetadata* pTrackMetadata, /*const*/TagLib::MP4::Tag& tag) {
+    if (kDebugMetadata) {
+        for (TagLib::MP4::ItemListMap::ConstIterator it(
+                tag.itemListMap().begin()); it != tag.itemListMap().end();
+                ++it) {
+            qDebug() << "MP4" << toQString((*it).first) << "-"
+                    << toQStringFirst((*it).second);
+        }
+    }
+
+    readTag(pTrackMetadata, tag);
+
+    // Get BPM
+    if (tag.itemListMap().contains("tmpo")) {
+        // Read the BPM as an integer value.
+        pTrackMetadata->setBpm(tag.itemListMap()["tmpo"].toInt());
```

Comment:
Above we allow here all sort of formats even though in is specified. I think we should do it here as well.